docs/custom-flows/google-one-tap.mdx

```diff
@@ -0,0 +1,145 @@
+---
+description: Learn how to build a custom Google One Tap authentication flow using Clerk's API.
+---
+
+# Build a custom Google One Tap authentication flow
+
+<Callout type="danger">
+  This guide is for users who want to build a *custom* user interface using the Clerk API. To authenticate users with Google One Tap using a *prebuilt* UI, you should use Clerk's [`<GoogleOneTap>`](/docs/components/authentication/google-one-tap) component.
+</Callout>
+
+[Google One Tap](https://developers.google.com/identity/gsi/web/guides/features) enables users to press a single button to authentication in your Clerk application with a Google account.
+
+This guide will walk you through how to build a custom Google One Tap authentication flow.
+
+<Steps>
+
+### Enable Google as a social connection
+
+To use Google One Tap with Clerk, you must [enable Google as a social connection in the Clerk Dashboard](/docs/authentication/social-connections/google#configuring-google-social-connection) and make sure to use custom credentials.
+
+### Create the Google One Tap authentication flow
+
+To authenticate user with Google One Tap, you must:
+
+1. Initialize a ["Sign In With Google"](https://developers.google.com/identity/gsi/web/reference/js-reference) client UI, passing in your Client ID.
+2. Use the response to authenticate the user in your Clerk app if the request was successful.
+3. Redirect the user back to the page they started the authentication flow from by default, or to another URL if necessary.
+
+The following example creates a component that implements a custom Google One Tap authentication flow, which can be used in a sign-in or sign-up page.
+
+<Tabs items={["Next.js"]}>
+  <Tab>
+  ```tsx filename="app/components/custom-one-tap.tsx"
```

Comment:
Best practice is to capitalize component names.

```suggestion
  ```tsx filename="app/components/CustomGoogleOneTap.tsx"
```